### PR TITLE
Center Bind Bonus: Two pawns controlling the same central square

### PR DESCRIPTION
**Pull request by zamar**
STC:  

LLR: 3.14 (-2.94,2.94) [-1.50,4.50]
Total: 15974 W: 3291 L: 3133 D: 9550

LTC: 

LLR: 3.24 (-2.94,2.94) [0.00,6.00]
Total: 10449 W: 1837 L: 1674 D: 6938

Idea from Lyudmil Tsvetkov.

### CODE REVIEW

File: src/pawns.cpp [hunk 1]
```diff
@@ -61,6 +61,14 @@ namespace {
   // Unsupported pawn penalty
   const Score UnsupportedPawnPenalty = S(20, 10);
 
+  // Center Bind Bonus: Two pawns controlling the same central square
+  const Bitboard CenterBindMask[] = {
```
Comment (mcostalba) [inline src/pawns.cpp]: Please use 

const Bitboard CenterBindMask[COLOR_NB]

to document is indexed by color

File: src/pawns.cpp [hunk 2]
```diff
@@ -195,6 +203,10 @@ namespace {
     b = e->semiopenFiles[Us] ^ 0xFF;
     e->pawnSpan[Us] = b ? int(msb(b) - lsb(b)) : 0;
 
+    // Center binds: Two pawns controlling the same central square
+    const Bitboard pawnBinds = shift_bb<Right>(ourPawns) & shift_bb<Left>(ourPawns);
```
Comment (mcostalba) [inline src/pawns.cpp]: I would reuse 'b' instead of intorducing

const Bitboard pawnBinds

This is how it is done in teh above code.

Comment (mcostalba) [inline src/pawns.cpp]: I would right directly:

b = shift_bb<Right>(ourPawns) & shift_bb<Left>(ourPawns) & CenterBindMask[Us];
score += popcount<Max15>(b) * CenterBindBonus;

File: src/pawns.cpp [hunk 3]
```diff
@@ -61,6 +61,14 @@ namespace {
   // Unsupported pawn penalty
   const Score UnsupportedPawnPenalty = S(20, 10);
 
+  // Center Bind Bonus: Two pawns controlling the same central square
+  const Bitboard CenterBindMask[] = {
+    (FileDBB | FileEBB) & (Rank5BB | Rank6BB | Rank7BB),
+    (FileDBB | FileEBB) & (Rank4BB | Rank3BB | Rank2BB)
+  };
+
+  const Score CenterBindBonus = S(16, 0);
```
Comment (mcostalba) [inline src/pawns.cpp]: Here we don't add bonus suffix, so i would call it:

const Score CenterBind = S(16, 0);

**Comment by mcostalba** [comment]
Congrats Joona and Lyudmil !

Joona, I have just added some comments in the code on naming and other small things.

**Comment by Nazine** [comment]
Joona is a horrible coder in terms of lucidity, meanwhile Costalba succeeds at it.

**Comment by mcostalba** [comment]
@Nazine you are about to be banned.

**Comment by glinscott** [comment]
Congrats!!  Marcos suggestions look good.  Merge any time :).

**Pull request closed without merging**